bats/sql-multi-db.bats

```diff
@@ -83,3 +83,122 @@ seed_repos_with_tables_with_use_statements() {
+    dolt init
+
+    run dolt sql << SQL
+CREATE DATABASE mydb;
+SHOW DATABASES;
+USE mydb;
+CREATE TABLE test (
+    pk int primary key
+);
+INSERT INTO test VALUES (222);
+SELECT COUNT(*) FROM test WHERE pk=222;
+DROP DATABASE mydb;
+SQL
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "dolt_repo_$$" ]] || false
+    [[ "$output" =~ "information_schema" ]] || false
+    [[ "$output" =~ "mydb" ]] || false
+    # From COUNT
+    [[ "$output" =~ "1" ]] || false
+
+    run dolt sql -q "SHOW DATABASES"
+    [[ ! "$output" =~ "mydb" ]] || false
+}
+
+@test "sql create and dropnew database" {
+    dolt init
+
+    run dolt sql << SQL
+CREATE DATABASE mydb;
+DROP DATABASE mydb;
+USE mydb;
+SQL
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "database not found: mydb" ]] || false
+}
+
+@test "sql create new database IF EXISTS works" {
+    dolt init
+
+    # Test bad syntax.
+    run dolt sql -q "CREATE DATABASE IF EXISTS test;"
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "Error parsing SQL" ]] || false
+
+    run dolt sql << SQL
+CREATE DATABASE IF NOT EXISTS test;
+SHOW DATABASES;
+USE test;
+CREATE TABLE test (
+    pk int primary key
+);
+INSERT INTO test VALUES (222);
+SELECT COUNT(*) FROM test WHERE pk=222;
+DROP DATABASE test;
+SQL
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "dolt_repo_$$" ]] || false
+    [[ "$output" =~ "information_schema" ]] || false
+    [[ "$output" =~ "test" ]] || false
+    # From COUNT
+    [[ "$output" =~ "1" ]] || false
+
+    run dolt sql << SQL
+CREATE DATABASE IF NOT EXISTS test;
+SHOW DATABASES;
+USE test;
+DROP DATABASE IF EXISTS test;
+DROP DATABASE IF EXISTS test;
+SQL
+    # The second drop database should just return a warning resulting in a status of 0.
+    [ "$status" -eq 0 ]
+
+    run dolt sql << SQL
+CREATE DATABASE IF NOT EXISTS test;
+SHOW DATABASES;
+USE test;
+DROP DATABASE IF NOT EXISTS test;
+SQL
+    # IF NOT should not work with drop.
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "Error parsing SQL" ]] || false
+}
+
+@test "sql drop database errors for non memory databases" {
+    dolt init
+
+    run dolt sql -q "DROP DATABASE dolt_repo_$$"
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "DROP DATABASE isn't supported for database dolt_repo_$$" ]] || false
+
+    run dolt sql -q "DROP DATABASE information_schema"
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "DROP DATABASE isn't supported for database information_schema" ]] || false
+}
+
+@test "sql create new database via SCHEMA alias" {
+    dolt init
+
+    run dolt sql << SQL
+CREATE SCHEMA mydb;
+SHOW DATABASES;
+USE mydb;
+CREATE TABLE test (
+    pk int primary key
+);
+INSERT INTO test VALUES (222);
+SELECT COUNT(*) FROM test WHERE pk=222;
+DROP SCHEMA mydb;
+SQL
+    [[ "$output" =~ "dolt_repo_$$" ]] || false
+    [[ "$output" =~ "information_schema" ]] || false
+    [[ "$output" =~ "mydb" ]] || false
+    [[ "$output" =~ "1" ]] || false
+
+    run dolt sql -q "SHOW DATABASES"
+    [[ ! "$output" =~ "mydb" ]] || false
+}
```

Comment:
I think we need some more bats test coverage here. Here are some ideas:
* non-happy path tests. drop a db that does not exist, create a db that already exists, etc.
* Add some querying across two databases, both created and the default
* I think SHOW DATABASES and USE need to be explicitly tested. Happy and non-happy.